ci.md

```diff
@@ -0,0 +1,201 @@
+# Ignition Continuous Integration
+
+When developers push code changes into Ignition, a series of automated tests
+run for various platforms. This process is called Continuous Integration, or CI.
+This document covers all the CI jobs that are available for Ignition and how to
+interpret their results.
+
+## Types of checks
+
+The main type of check that is performed by CI is compiling the code and running
+all automated tests. Some jobs may also run linters and other static checkers.
+Finally, we have checks that add labels, check DCO and coverage.
+
+All checks are visible at the bottom of a pull request.
+
+![PR checks](images/PR_checks.png)
+
+We use two platforms to run CI, each of them running different builds.
+
+### Jenkins server
+
+Our Jenkins server is located at https://build.osrfoundation.org/. It runs builds
+for all supported operating systems. It also runs an ABI checker for all stable
+branches.
+
+Each pull request to a stable or `main` branch will trigger:
+
+* `<library>-ci-pr_any-ubuntu-auto-amd64`: Build and test on Ubuntu Linux
+    * Builds are run inside Docker containers for Bionic or Focal.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-homebrew-amd64`: Build and test on macOS
+    * Dependencies are installed from homebrew.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-windows*-amd64`: Build and test on Windows
+    * External dependencies are installed from vcpkg
+    * Library being tested and its Ignition dependencies are compiled using colcon.
+
+Pull requests to stable branches also trigger:
+
+* `<library>-abichecker-any_to_any-ubuntu_auto-amd64`: Run ABI checker
+
+### GitHub Actions
+
+GitHub actions run on GitHub's servers. Currently, it builds Linux for all
+libraries, and macOS for some of them.
+
+Every commit to any branch will trigger:
+
+* `Ubuntu CI / Ubuntu Bionic CI (push)`: Build, test and run linters on Ubuntu Bionic.
+    * Builds are run inside Docker containers.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+    * Test coverage is computed and uploaded to Codecov.
```

Comment:
Is test coverage only for Bionic and not Focal?
The Bionic and Focal items are exactly the same other than the distro name. It felt dizzy reading this dense technical text twice. Is it possible to combine them?

Reply:
Yeah the only difference is that we only run coverage for one of them, because we can't keep track of multiple coverage results for the same branch. I agree that combining should make it easier to read :+1: